dist/dialog.linker.build.js

```diff
@@ -71,17 +78,21 @@ window.jQuery = window.jQuery || window.shoestring;
 		this.$background.attr( "id", this.$el.attr( "id" ) + "-background" );
 		this._setBackgroundTransparency();
 
-		if( this.isSetScrollPosition() ) {
-			this.scroll = "pageYOffset" in w ? w.pageYOffset : ( docElem.scrollY || docElem.scrollTop || ( body && body.scrollY ) || 0 );
-			this.$el[ 0 ].style.top = this.scroll + "px";
-		} else {
-			this.$el[ 0 ].style.top = '';
-		}
+		this.scroll = "pageYOffset" in w ? w.pageYOffset : ( docElem.scrollY || docElem.scrollTop || ( body && body.scrollY ) || 0 );
+		this.$el[ 0 ].style.top = this.scroll + "px";
 
 		$html.addClass( cl.open );
 		this.isOpen = true;
 
-		window.location.hash = this.hash;
+		var cleanHash = w.location.hash.replace( /^#/, "" );
+		var lastHash = w.location.hash.split( "#" ).pop();
+
+		if( cleanHash.indexOf( "-dialog" ) > -1 && lastHash !== this.hash ){
+			w.location.hash += "#" + this.hash;
+		}
+		else if( lastHash !== this.hash ) {
+			w.location.hash = this.hash;
+		}
```

Comment:
Nit: can we just append the hash whenever it's not the same as the `lastHash`? Then we don't have to worry about the corner case where `-dialog` somehow ends up in the hash for other reasons.

That is, it seems like the fact that `-dialog` is in the hash shouldn't matter?